cmd/erasure-multipart.go

```diff
@@ -283,6 +283,17 @@ func (er erasureObjects) newMultipartUpload(ctx context.Context, bucket string,
 		parityBlocks = er.defaultParityCount
 	}
 
+	ecOrg := parityBlocks
+	for _, disk := range onlineDisks {
+		di, err := disk.DiskInfo(ctx)
+		if (err != nil || di.ID == "") && parityBlocks < len(onlineDisks)/2 {
+			parityBlocks++
```

Comment:
We should make sure that parityBlocks never grow beyond `len(onlineDisks)/2`  if they do reset them back to at least the maximum `len(onlineDisks)/2`

Reply:
@harshavardhana That is what  `parityBlocks < len(onlineDisks)/2` does.